subsys/fs/nvs/nvs.c

```diff
@@ -635,6 +635,14 @@ static int nvs_startup(struct nvs_fs *fs)
 			fs->data_wra += last_ate.offset;
 			fs->data_wra += nvs_al_size(fs, last_ate.len);
 
+			/* check if existing NVS entry is properly aligned according to
+			 * the current flash driver requirements
+			 */
+			if (fs->data_wra & (fs->flash_parameters->write_block_size - 1)) {
```

Comment:
There should probably be some `LOG_ERR` here as with the other `-EINVAL` (in `nvs_init`)